internal/pkg/config/x509_test.go

```diff
@@ -15,30 +16,52 @@
 package config
 
 import (
+	"path/filepath"
+	"strings"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
 
+var testJSONFileName = "./res/pkisetup-test.json"
+
 func TestLoadX509Config(t *testing.T) {
 	tests := []struct {
 		name        string
 		file        string
 		expectError bool
 	}{
-		{"509LoadOK", "./res/pkisetup-test.json", false},
+		{"509LoadOK", testJSONFileName, false},
 		{"509NonexistentFile", "./res/pkisetup-nope.json", true},
 		{"509InvalidJSON", "./res/pkisetup-fail.json", true},
 	}
+	assert := assert.New(t)
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			_, err := NewX509Config(tt.file)
-			if err != nil && !tt.expectError {
-				t.Errorf("unexpected error %v", err)
-				return
+			x509Config, err := NewX509Config(tt.file)
+
+			if !tt.expectError {
```

Comment:
this would read better as 

```go
			if tt.expectError {
				assert.NotNil(err, "expected error, none returned")
				assert.Empty(x509Config)
                        } else {
				assert.Nil(err, "unexpected error %v", err)
				assert.NotEmpty(x509Config)
                        }
```

Reply:
good point